foodgroup/chat_nav.go

```diff
@@ -102,14 +108,14 @@ func (s ChatNavService) CreateRoom(_ context.Context, sess *state.Session, inFra
 		room.Name = name
 
 		if err := s.chatRoomManager.CreateChatRoom(room); err != nil {
-			return wire.SNACMessage{}, fmt.Errorf("unable to create chat room: %w", err)
+			s.logger.Error(fmt.Sprintf("%s: %s", ErrChatNavRoomCreateFailed.Error(), err.Error()))
```

Comment:
It looks like your intent here is to capture both `ErrChatNavRoomCreateFailed` and the err returned by `CreateChatRoom`. That is a good idea.

Go provides an idiomatic way of doing this, which is called is error wrapping. This ensures that the two errors, in addition to any previous errors wrapped further up the chain, are included in a single error message. It also ensures that `errors.Is()` works.

```go
		if err := s.chatRoomManager.CreateChatRoom(room); err != nil {
			err = fmt.Errorf("%w: %w", ErrChatNavRoomCreateFailed, err)
			return wire.SNACMessage{}, err
		}
```

In addition, this error is already logged by the caller, so logging errors here is redundant.